From f09d97ed050da77ce9577f2d2764e42c8bb63e36 Mon Sep 17 00:00:00 2001
From: George McCollister <george.mccollister@gmail.com>
Date: Tue, 13 Oct 2015 14:52:10 -0500
Subject: [PATCH 2/2] Fix configure paths

Based on configure_var.patch from meta-openstack mod-wsgi_3.4

Signed-off-by: George McCollister <george.mccollister@gmail.com>
---
 configure.ac | 39 +++++++++++++++++++++++++++++++++++++++
 1 file changed, 39 insertions(+)

diff --git a/configure.ac b/configure.ac
index ab6145d..9ea3daa 100644
--- a/configure.ac
+++ b/configure.ac
@@ -100,9 +100,11 @@ fi
 
 AC_SUBST(PYTHON)
 
+if test -z "${PYTHON_VERSION}"; then
 PYTHON_VERSION=`${PYTHON} -c 'from sys import stdout; \
     from distutils import sysconfig; \
     stdout.write(sysconfig.get_config_var("VERSION"))'`
+fi
 
 PYTHON_LDVERSION=`${PYTHON} -c 'from sys import stdout; \
     from distutils import sysconfig; \
@@ -112,14 +114,22 @@ if test x"${PYTHON_LDVERSION}" = x""; then
     PYTHON_LDVERSION=${PYTHON_VERSION}
 fi
 
+if test -z "${PYTHON_INCLUDEPY}"; then
 CPPFLAGS1=`${PYTHON} -c 'from sys import stdout; \
     from distutils import sysconfig; \
     stdout.write("-I" + sysconfig.get_config_var("INCLUDEPY"))'`
+else
+CPPFLAGS1="${PYTHON_INCLUDEPY}"
+fi
 
+if test -z "${PYTHON_CFLAGS}"; then
 CPPFLAGS2=`${PYTHON} -c 'from sys import stdout; \
     from distutils import sysconfig; \
     stdout.write(" ".join(filter(lambda x: x.startswith("-D"), \
     sysconfig.get_config_var("CFLAGS").split())))'`
+else
+CPPFLAGS2="${PYTHON_CFLAGS}"
+fi
 
 if test "${ENABLE_EMBEDDED}" != "yes"; then
 CPPFLAGS3="-DMOD_WSGI_DISABLE_EMBEDDED"
@@ -131,22 +141,43 @@ CPPFLAGS="${CPPFLAGS} ${CPPFLAGS1} ${CPPFLAGS2} ${CPPFLAGS3}"
 
 AC_SUBST(CPPFLAGS)
 
+if test -z "${PYTHON_LIBDIR}"; then
 PYTHONLIBDIR=`${PYTHON} -c 'from sys import stdout; \
     from distutils import sysconfig; \
     stdout.write(sysconfig.get_config_var("LIBDIR"))'`
+else
+PYTHONLIBDIR="${PYTHON_LIBDIR}"
+fi
+if test -z "${PYTHON_CFGDIR}"; then
 PYTHONCFGDIR=`${PYTHON} -c 'from sys import stdout; \
     import distutils.sysconfig; \
     stdout.write(distutils.sysconfig.get_python_lib(plat_specific=1, \
     standard_lib=1) +"/config")'`
+else
+PYTHONCFGDIR="${PYTHON_CFGDIR}"
+fi
+if test -z "${PYTHON_FRAMEWORKDIR}"; then
 PYTHONFRAMEWORKDIR=`${PYTHON} -c 'from sys import stdout; \
     from distutils import sysconfig; \
     stdout.write(sysconfig.get_config_var("PYTHONFRAMEWORKDIR"))'`
+else
+PYTHONFRAMEWORKDIR="${PYTHON_FRAMEWORKDIR}"
+fi
+
+if test -z "${PYTHON_FRAMEWORKPREFIX}"; then
 PYTHONFRAMEWORKPREFIX=`${PYTHON} -c 'from sys import stdout; \
     from distutils import sysconfig; \
     stdout.write(sysconfig.get_config_var("PYTHONFRAMEWORKPREFIX"))'`
+else
+PYTHONFRAMEWORKPREFIX="${PYTHON_FRAMEWORKPREFIX}"
+fi
+if test -z "${PYTHON_FRAMEWORK}"; then
 PYTHONFRAMEWORK=`${PYTHON} -c 'from sys import stdout; \
     from distutils import sysconfig; \
     stdout.write(sysconfig.get_config_var("PYTHONFRAMEWORK"))'`
+else
+PYTHONFRAMEWORK="${PYTHON_FRAMEWORK}"
+fi
 
 if test "${PYTHON_LDVERSION}" != "${PYTHON_VERSION}"; then
     PYTHONCFGDIR="${PYTHONCFGDIR}-${PYTHON_LDVERSION}"
@@ -170,12 +201,20 @@ if test "${PYTHONFRAMEWORKDIR}" = "no-framework" -o \
         LDLIBS1="-lpython${PYTHON_VERSION}"
     fi
 
+    if test -z "${PYTHON_LIBS}"; then
     LDLIBS2=`${PYTHON} -c 'from sys import stdout; \
         from distutils import sysconfig; \
         stdout.write(sysconfig.get_config_var("LIBS"))'`
+    else
+    LDLIBS2="${PYTHON_LIBS}"
+    fi
+    if test -z "${PYTHON_SYSLIBS}"; then
     LDLIBS3=`${PYTHON} -c 'from sys import stdout; \
         from distutils import sysconfig; \
         stdout.write(sysconfig.get_config_var("SYSLIBS"))'`
+    else
+    LDLIBS3="${PYTHON_SYSLIBS}"
+    fi
 else
     LDFLAGS1="-Wl,-F${PYTHONFRAMEWORKPREFIX} -framework ${PYTHONFRAMEWORK}"
 
-- 
2.4.5

